### PR TITLE
Django3

### PR DESCRIPTION
**Pull request by yychen**
Major upgrade to Django3.

### CODE REVIEW

File: src/events/tests/test_admin.py [hunk 1]
```diff
@@ -17,12 +18,21 @@
 def choices_factory():
 
     def build_changelist_choices(request, model_class, modeladmin, index=0):
+        request.user = AnonymousUser()
+        # yychen 2020-02-02: the request factory does not support middleware,
+        # so no user would be set in request.
+        # However, a check of request.user is performed in ChangeList()
+        # which will cause error.
+        # So we inject AnonymousUser here
+        # https://stackoverflow.com/questions/54183451/django-unit-testing-attributeerror-wsgirequest-object-has-no-attribute-user
```
Comment (uranusjr) [inline src/events/tests/test_admin.py]: Can be shortened to https://stackoverflow.com/questions/54183451

File: src/proposals/tests/test_management.py [hunk 2]
```diff
@@ -215,6 +221,7 @@ def test_command_with_slack():
         slack=True,
     )
     assert Slack.notify.call_count == 1
+    Slack.notify = _restore
```
Comment (uranusjr) [inline src/proposals/tests/test_management.py]: Can we use `with unittest.mock.patch.object(Slack, 'notify', MagicMock(...))` instead?

File: src/users/urls.py [hunk 3]
```diff
@@ -22,12 +28,16 @@
...
+                                       template_name='registration/password_reset.html',
+                                       email_template_name='registration/password_reset_email.txt'),
+        name='password_reset'),
     url(r'^password-reset/done/$',
         views.password_reset_done, name='password_reset_done'),
     url(r'^password-reset/(?P<uidb64>[0-9A-Za-z_\-]+)/'
         r'(?P<token>[0-9A-Za-z]{1,13}-[0-9A-Za-z]{1,20})/$',
-        views.password_reset_confirm, name='password_reset_confirm'),
+        auth.PasswordResetConfirmView.as_view(form_class=SetPasswordForm),
+        name='password_reset_confirm'),
```
Comment (uranusjr) [inline src/users/urls.py]: Maybe put these long view declarations in `views.py`? Like

https://github.com/pycontw/pycon.tw/blob/a8a519b2c0dbd6717c228f7c1e11f5082697a9d4/src/core/views.py#L67-L68

Comment (yychen) [reply src/users/urls.py]: 豪!

File: src/proposals/tests/test_management.py (hunk 2 above)
Comment (yychen) [reply src/proposals/tests/test_management.py]: 太棒了！好喔！

**Comment by yychen** [comment]
感恩 TP m(_ _)m

**Pull request merged**